### ISSUE TITLE
NixOS website from a non-developer's perspective

### ISSUE THREAD

**Issue by Raikiri** [NixOS/nixos-homepage#828]
**Introduce your stance**
I consider myself to be a Nix/NixOS admirer, but I don't use NixOS for development. I want to use it as my daily driver because I love the way it allows me to declaratively configure my system and the way Nix handles software dependencies. 

**Describe the issue**
The problem is that after reading a bunch of articles on NixOS I got really excited, but the homepage tried very hard to make it looks like NixOS has nothing to do with me and has none of the stuff that I actually want. Just look at all of its main pages:

**Reproducible builds and deployments.**
![image](https://user-images.githubusercontent.com/1657728/164162332-15eead16-cf88-4b90-8eaf-275c0dc94aec.png)
The first thing a user sees on the main page. I'm not planning to use it for development. Heck, I might not even know what "reproducible build" is or what "deployments" is. I just want to have a sane way to manage my desktop machine without imperative nonsense.

**Reproducible Declarative Reliable**
![image](https://user-images.githubusercontent.com/1657728/164162249-782a859c-5a19-48fd-943c-b06ccd3c9cab.png)
thank god, at least these terms are explained for a layman and this was literally the only thing that convinced me I'm actually on the right website.

**Explore**
![image](https://user-images.githubusercontent.com/1657728/164162801-c4149050-b63e-419b-97cc-6257afcab3a3.png)
1) A happy lion that sets up an always working development environment, yay! Great for him. Too bad I don't care about this thing, let's just scroll down and find something more relevant to me.
![image](https://user-images.githubusercontent.com/1657728/164162940-42ff44ed-973f-4a7e-bd37-2b18fa0643f0.png)
2) One tool, many languages -- again, yay for happy octopus but I don't even plan to use any languages, why are we talking about a tool for that?
![image](https://user-images.githubusercontent.com/1657728/164163055-1c76354f-4b83-4526-b06e-690f1670b5e8.png)
3) I don't even know what CI is, but yay! (completely irrelevant to me again)
![image](https://user-images.githubusercontent.com/1657728/164163114-72e87377-47c7-4839-8b4e-aa472d960d28.png)
4) From zero to what cloud? For me "cloud" means where I backup my photos. And this thing is talking Amazon somethi-thing and that other DevOps stuff that I have nothing to do with.

**Learn**
![image](https://user-images.githubusercontent.com/1657728/164162537-3216c9dd-8971-4ef7-b72b-dc2a2286ca27.png)
"First steps in Nix" leads to how to set up a developer's environment. Again, completely irrelevant to a normie user. Already made an issue to that https://github.com/NixOS/nixos-homepage/issues/825 (mark that one as a duplicate? because the issue is throughout the whole website, really, not just that section)

**Additional context**
This website looks and feels really cool, but it's clearly made for a type of user that I'm absolutely not. And it makes me feel very unwelcome and gives an impression that I'm not a target audience at all. However, after using NixOS for a while I really dig it and I really feel that I am. And I want less people to be intimidated by its website.

I feel this thing is part of the overall UX for a new user that wants to give NixOS a try, and it's a pretty important part. This is a also definitely a recurring theme throughout the internet : people have no idea why they want to try NixOS after reading any of the user-facing advertisements on the website and any of the documentation that's right on the surface. And they get way more interested after reading those random impressions blogposts from other users and watching some youtube videos. Why does it have to be this way, totally backwards?

**Comment by fricklerhandwerk** [comment]
The [Marketing team](https://nixos.org/community/teams/marketing.html) decided in 2020 to [primarily target developers](https://discourse.nixos.org/t/marketing-team-meeting-minutes-3/6727) in a first round of improving the onboarding experience. See the other [Marketing team meeting notes](https://discourse.nixos.org/c/governance/marketing/28) for more details on the process.

I agree there is a severe lack of non-developer introductory material. Resources are limited, and there is much to do to substantially improve the situation for any major user group. Without having been involved, I presume it was the most reasonable decision to focus on developers, as they were already the major user group and also the most probable to contribute to further improving the ecosystem.

[Summer of Nix 2022](https://discourse.nixos.org/t/summer-of-nix-2022/18479) has a budget for improving communication, and is open for participant applications as well as project proposals. I would very much like to see people engaging to make Nix and its surrounding tools more approachable.

**Comment by Raikiri** [comment]
@fricklerhandwerk unfortunately it feels like currently the target audience of the main page advertisement is not just developers, but it's specifically DevOps and even more specifically, DevOps weirdly enough already know why they want Nix. So it's very self-affirming if I may say.

It's such a sad situation because I feel like it's really a case of chicken-and-egg problem where the nix infrastructure just needs more users to become more popular to attract more developers to become more user-friendly to attract more users. Like as if the whole thing needs just a kickstart because the foundation is truly profound even though most people don't know it's already there.

I try to contribute the way I feel most appropriate -- by advertising it among friends who might dig it. But at the same time I find it hard to convince them that: "don't worry about the homepage, it's a little bit weird, but it's actually awesome even if you're not a DevOps engineer".

**Comment by vlinkz** [comment]
I agree that the NixOS landing page is very intimidating. With a new graphical installer (https://github.com/NixOS/nixpkgs/pull/161788) on the way, I think a site talking about deployments, containers, and such could scare users away. Not sure what the best solution would be, maybe have one site for Nix as a development and deployment tool, and make something new for NixOS as a desktop operating system? (Although NixOS is still far from being a user friendly desktop experience for non-technical users, for now)

**Comment by Raikiri** [comment]
I personally think that it's perfectly fine to aim the website at linux technical users for now. Because obviously some text configuration will be required. The best fit is probably the current arch audience, in my view.

One actionable change I would suggest is instead of advertising technical terms like "Declarative", advertise why it's good and then also mention what it's called. For example:
`Manage your system with one config file!
Nix is *declarative* which allows you to define its state in a single file rather than installing and removing packages via command line. You can easily transfer it to your other machines too!`

`Dependency conflicts are gone for good!
With Nix's innovative way of tracking package versions and providing dependencies, each software package uses exactly the version that it was compiled and tested with.`

Advantages like this are definitely going to get my mom excited. But it's too early to target her. Instead, any arch user will and should get excited about obviously good stuff like this.

**Comment by misuzu** [comment]
My biggest complain about nixos.org website is that it's completely unrelated to NixOS and looks more like nix.org.

**Comment by Raikiri** [comment]
> The solution is to create a new website for Nix and use nixos.org to focus on the OS. That was discussed multiple times in marketing meetings, it's just that no one has done it yet.

Unfortunately I imagine "make a separate website" is not a particularly actionable task in any foreseeable future and it's more constructive to be find near-term solutions how to make do with what we have. I'd be glad to be proven wrong here though.

**Comment by timfenney** [comment]
If you want to reach developers, I would look to projects like Ruby on Rails. DHH captured the imagination of web devs by just _showing them you can build a blog in 15 minutes_.

This is what you see on the home page:
![Screenshot from 2023-02-02 21-20-58](https://user-images.githubusercontent.com/654400/216511767-47e41a59-2e0a-45ed-95c8-b046b9c5ba16.png)

What would be great is a similar set of demos with Nix and NixOS, to target the developer crowd.
 
I agree with OP's points. A lot of top distributions on Distrowatch give feature tours and show the polish of the initial experience. Rich Hickey gave a talk on Simplicity, which he demonstrates is better than ease, but easy is pretty good too. Asking someone to install Linux is getting harder in the age of WSL, Apple, and Codespaces, etc., so in future I don't think it makes sense for distributions to be as hard to use as Ubuntu.

One of the Clojurescript guys was joking about a metric, Time To First XML, and showed that the Clojure(script) home page was scarier than it needed to be and was probably driving users away.

In the end, people don't choose things for rational reasons most times. We are evolved to pick beautiful juicy fruit, and our products and documents need to show empathy for that.
